pytest_inmanta/plugin.py

```diff
@@ -615,6 +656,30 @@ def get_context_for(
             )
         return self.results[resources[0]]
 
+    def get_resource(
+        self, resource_type: str, **filter_args: object
+    ) -> typing.Optional[Resource]:
+        """
+        Get a resource of the given type and given filter on the results attributes from a Result object.
+        If multiple resource match, the first one is returned. If none match, None is returned.
+
+        :param resource_type: The exact type used in the model (no super types)
+        """
+        return get_resource(self.results.keys(), resource_type, **filter_args)
+
+
+class DeployResult(Result):
```

Comment:
```suggestion
DeployResult = Result
```

I think this is more robust wrt to isinstance checks ?

Reply:
Good point, we need to be backwards compatible in two directions.